Formula/commandbox.rb

```diff
@@ -19,13 +19,13 @@ class Commandbox < Formula
   depends_on "openjdk@11"
 
   resource "apidocs" do
-    url "https://downloads.ortussolutions.com/ortussolutions/commandbox/5.3.1/commandbox-apidocs-5.3.1.zip"
-    sha256 "2090d90b7f53e4e776178d55ac12ae2fc5ec3d05a217cc60f02587a338f306cf"
+    url "https://downloads.ortussolutions.com/ortussolutions/commandbox/5.4.0/commandbox-apidocs-5.4.0.zip"
+    sha256 "356b262a9bc388c491db54d6af9200f7653e0225cc86cd91ae430297ce0f01e3"
   end
 
   def install
     (libexec/"bin").install "box"
-    (bin/"box").write_env_script libexec/"bin/box", Language::Java.overridable_java_home_env("11")
```

Comment:
I believe we dont want to use `java_home_env` (checkout PRs with [this deprecate-java-home-cmd label](https://github.com/Homebrew/homebrew-core/labels/deprecate-java-home-cmd))

Reply:
This is fine; what we don't want to use is `java_home_cmd`

Reply:
@chenrui333 I'm not clear what you're suggesting here.  My understanding is the line above sets the Java home to ensure the correct version is set.  If we remove that, what will make sure that `box` is run with the correct Java version?

Reply:
Sooo..... are there changes needed or not?  I'm still not clear on exactly what is being suggested.  @chenrui333 is saying something needs to change and @carlocab is saying it's fine the way it is.

Reply:
No, this works fine. However, I am wondering why you've replaced `overrideable_java_home_env` with `java_home_env`.

Reply:
@carlocab The reason I switched is because when a user had an explicit java_home set to an incompatible version of Java, brew wouldn't force the correct Java home env var which broke my tool.  At least that's what appeared to be happening.  From what I could find, there's no docs on this stuff and zero comments in the code so I was just grasping at the proper fix.  I do know this-- Java 16 is a bit fat no-go so even if the user's PC has an explicit `JAVA_HOME` env var pointing to Java 16, I need Brew to force Java 11 instead or that setting is pointless.

Reply:
@carlocab FYI, here's the source code (with no comments) for those methods
https://github.com/Homebrew/brew/blob/04e09de1d84906d97230fe305a492aa802f4f9d9/Library/Homebrew/language/java.rb#L39-L45
You can see one of them is using a bash expansion that only uses the specific java home as a default if there isn't already an env var set.  But in this case, I don't _want_ my Java home to be overridable.  I want it to be Java 11 for certain and no higher.

Reply:
> The reason I switched is because when a user had an explicit java_home set to an incompatible version of Java, brew wouldn't force the correct Java home env var which broke my tool.

This is actually the preferred approach for most upstream projects that we encounter. That's why we're all a bit surprised 😄 . But in that case it's the correct change.